src/test/java/com/atlassian/bitbucket/jenkins/internal/status/BitbucketBuildStatusFactoryTest.java

```diff
@@ -19,98 +19,150 @@
 
 import static org.hamcrest.CoreMatchers.*;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
 
 @RunWith(MockitoJUnitRunner.class)
 public class BitbucketBuildStatusFactoryTest {
 
-    private static final String BUILD_DISPLAY_NAME = "#15";
     private static final String BUILD_DURATION = "400 secs";
+    private static final String BUILD_ID = "15";
+    private static final String BUILD_DISPLAY_NAME = "#" + BUILD_ID;
     private static final String BUILD_URL = "http://www.example.com:8000";
-    private static final String PROJECT_NAME = "Project Name";
+    private static final String FREESTYLE_PROJECT_NAME = "Freestyle Project";
+    private static final String WORKFLOW_JOB_NAME = "branch-name";
+    private static final String WORKFLOW_PROJECT_NAME = "MultiBranch Project";
 
     @Rule
     public final JenkinsRule jenkinsRule = new JenkinsRule();
+    private BitbucketBuildStatusFactory buildStatusFactory;
     @Mock
-    private AbstractBuild build;
+    private BitbucketCICapabilities ciCapabilities;
+    @Mock
+    private Run freeStyleRun;
+    private Job freeStyleProject;
     @Mock
     private Jenkins parent;
     @Mock
-    private Project project;
+    private Run workflowRun;
```

Comment:
nit/tip/personal pet peeve: I like to have the `?` generics for these types to remove the generic type warnings (thanks Java's horrendous generic type system), so I use `Run<?, ?>`, etc. 
One side-effect of it is that some of the mockito `when()...` calls will fail because Mockito can't understand the complex generics, so then the solution is to use the roundabout mocking style, e.g.:
```
doReturn(project).when(build).getParent();
```

Reply:
I also hate those warnings. Happy to refactor.

Reply:
Okay just tried this and suddenly realized if we add <?, ?> to our run, then the job that's being added as a parent depends on that capture (and it doesn't capture ?, it captures whatever the run extends). Is there a way around this?

Reply:
I was able to get it to compile and tests to run by changing line 57 to `doReturn(freeStyleProject).when(freeStyleRun).getParent();`

That said, as much as I hate warnings, I'm also not sure if I like `<?, ?>` any more... ¯\_(ツ)_/¯

Reply:
Digging into the type system a bit more, I was also able to get the tests to run by declaring the jobs more specifically:
```
    @Mock
    private Run<FreeStyleProject, ?> freeStyleRun;
    private FreeStyleProject freeStyleProject;
    [...]
    @Mock
    private Run<WorkflowJob, ?> workflowRun;
    private WorkflowJob workflowJob;
```

That means you get to use the more natural mocking style:
```
        when(freeStyleRun.getParent()).thenReturn(freeStyleProject);
```
and
```
        when(workflowRun.getParent()).thenReturn(workflowJob);
```

That said, all the above is a nit. I think this is something we can let pass.